File: wtf-android/src/main/java/wtf/api/ApiUiClose.java
package wtf.api;

import wtf.sdk.JSO;
import wtf.sdk.WtfApi;
import wtf.sdk.WtfCallback;
import wtf.sdk.WtfHandler;
import wtf.sdk.WtfUi;

public class ApiUiClose extends WtfApi {

    @Override
    public WtfHandler getHandler() {
        return new WtfHandler() {

            @Override
            public void onCall(JSO jso, final WtfCallback responseCallback) {
                WtfUi ui = getCallerUi();

                ui.closeUi(jso);

                if (null != responseCallback) responseCallback.onCallBack(jso);
            }
        };
    }
}

File: wtf-ios/demo/ApiUiClose.m
#import "ApiUiClose.h"
#import "JSO.h"

@implementation ApiUiClose

- (WtfHandler) getHandler{
    return ^(JSO *jso, WtfCallback responseCallback) {
        dispatch_async(dispatch_get_main_queue(), ^{
            
            WtfUi ui = self.currentUi;
            
            [ui finishUi];
            
            //use responseCallback
            if(responseCallback)
            responseCallback([JSO id2o:@{@"STS":@"OK"}]);
        });
    };
}

@end
